src/tools/tidy/src/fluent_alphabetical.rs

```diff
@@ -24,30 +26,29 @@ fn check_alphabetic(
     check: &mut RunningCheck,
     all_defined_msgs: &mut HashMap<String, String>,
 ) {
-    let mut matches = message().captures_iter(fluent).peekable();
-    while let Some(m) = matches.next() {
-        let name = m.get(1).unwrap();
-        if let Some(defined_filename) = all_defined_msgs.get(name.as_str()) {
-            check.error(format!(
-                "{filename}: message `{}` is already defined in {defined_filename}",
-                name.as_str(),
-            ));
-        }
+    let resource = parser::parse(fluent).expect("Errors encountered while parsing a resource.");
```

Comment:
It would be nice to report in which filename the error was encountered, otherwise it's very hard to tell.